[tar] 1.35 fails to build for 32-bit msys2

I know 32-bit msys2 is no longer supported...

```
checking for gcc option for timestamps after 2038... support not detected
configure: error: in `/t/r/tar/src/tar-1.35':
configure: error: could not enable timestamps after mid-January 2038.
This package recommends support for these later
timestamps. However, to proceed with signed 32-bit
time_t even though it will fail then, configure with
'--disable-year2038'.
```

Just wanted to record this somewhere.  I will look into whether there is some define to get 64-bit `time_t` on 32-bit msys2 (somehow I doubt there is if configure didn't find it), or if that configure option needs to be added for i686.

--- a/tar/PKGBUILD
+++ b/tar/PKGBUILD
@@ -2,7 +2,7 @@
 
 pkgname=tar
 pkgver=1.35
-pkgrel=1
+pkgrel=2
 pkgdesc="Utility used to store, backup, and transport files"
 arch=('i686' 'x86_64')
 url="https://www.gnu.org/software/tar/tar.html"
@@ -37,12 +37,19 @@ build() {
 
   export gl_cv_have_weak=no
   local CYGWIN_CHOST="${CHOST/-msys/-cygwin}"
+  local -a extra_config
+  # 32-bit cygwin only has 32-bit time_t
+  # https://github.com/msys2/MSYS2-packages/issues/4078
+  if [[ "$CARCH" == "i686" ]]; then
+    extra_config+=("--disable-year2038")
+  fi
   ./configure \
     --build=${CYGWIN_CHOST} \
     --prefix=/usr \
     --sbindir=/usr/bin \
     --libexecdir=/usr/lib/tar \
-    --enable-backup-scripts
+    --enable-backup-scripts \
+    "${extra_config[@]}"
   make
 }
 

